### PR TITLE
Dynamic widget icons for scheduling email

### PR DESCRIPTION
**Pull request by oleg-morozov**
JIRA: TNT-574

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7501/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9361/ : FAILURE in 4m 31s

File: libs/sdk-ui-kit/src/Icon/icons/InsightIcons/TreeMap.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    const sizes: {
+        width?: number;
+        height?: number;
+    } = {};
+
+    width && (sizes.width = width);
+    height && (sizes.height = height);
+
+    return (
+        <svg {...sizes} className={className} viewBox="0 0 26 26" xmlns="http://www.w3.org/2000/svg">
```
Comment (scavnickyj) [inline libs/sdk-ui-kit/src/Icon/icons/InsightIcons/TreeMap.tsx:18]: isn't this the same as just doing `<svg width={width} height={height} />` as the props are optional?

File: libs/sdk-model/src/dashboard/analyticalWidgets.ts [hunk 2]
```diff
@@ -108,3 +108,10 @@ export interface IInsightWidget extends IInsightWidgetBase, IDashboardObjectIden
  * @alpha
  */
 export interface IInsightWidgetDefinition extends IInsightWidgetBase, Partial<IDashboardObjectIdentity> {}
+
+/**
+ * @alpha
+ */
+export interface IInsightWidgetWithVisType extends IInsightWidget {
+    visualizationUrl: string | undefined;
+}
```
Comment (scavnickyj) [inline libs/sdk-model/src/dashboard/analyticalWidgets.ts:117]: Can we keep this extension contained in the component itself?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7508/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9368/ : FAILURE in 3m 12s

File: libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/AttachmentsSelectionDropdown.tsx [hunk 3]
```diff
@@ -57,6 +57,31 @@ const AttachmentsSelectionDropdownComponent: React.FC<IAttachmentsSelectionDropd
...
+            "local:pushpin": <Icon.Geo {...ICON_PROPS} />,
+            "local:pie": <Icon.Pie {...ICON_PROPS} />,
+        };
+
+        if (widget.visualizationUrl) {
+            return map[widget.visualizationUrl];
+        }
+        return null;
+    };
+
```
Comment (scavnickyj) [inline libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/AttachmentsSelectionDropdown.tsx:84]: We could move this to ui-kit as a more general component.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7519/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9381/ : FAILURE in 14m 28s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7521/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9383/ : FAILURE in 13m 57s

**Comment by oleg-morozov** [comment]
extended test

**Comment by oleg-morozov** [comment]
extended check sonar

**Comment by oleg-morozov** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7525/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/195/ : SUCCESS in 10m 07s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7522/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9384/ : SUCCESS in 13m 52s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR2418

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7524/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1286/ : SUCCESS in 16m 41s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1447/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7523/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1447/ : UNSTABLE in 22m 35s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7529/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9389/ : SUCCESS in 14m 28s

**Review by tomassatinsky — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2418

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2179/ : SUCCESS in 1m 18s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2465/ : SUCCESS in 27s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1289/ : SUCCESS in 19m 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9391/ : SUCCESS in 14m 25s

**Pull request merged**